### DISCUSSION TITLE
Single cell selection / click in table

### DISCUSSION THREAD

**Discussion by mbolder** [zauberzeug/nicegui#979] [Q&A] [answered]
### Question

Hello, I have implemented a table with Postgres data and would love be able to select specific cell values in that table, 
I can get the values of a row with the tickbox and on_select method, but that gives me a row. Is there a way to get a cell value ?

**Comment by falkoschindler** [comment, marked as answer]
Good question, @mbolder!

Clicking cells doesn't seem to be natively supported by Quasar's [QTable](https://quasar.dev/vue-components/table) component. But we can overwrite the "body" slot and use a custom "cell-click" event:

```py
columns = [
    {'name': 'name', 'label': 'Name', 'field': 'name'},
    {'name': 'age', 'label': 'Age', 'field': 'age'},
]
rows = [
    {'name': 'Alice', 'age': 18},
    {'name': 'Bob', 'age': 21},
    {'name': 'Carol'},
]

table = ui.table(columns=columns, rows=rows, row_key='name').classes('w-72')
table.add_slot('body', r'''
    <q-tr :props="props">
        <q-td
            v-for="col in props.cols"
            :key="col.name"
            :props="props"
            @click="$parent.$emit('cell-click', {row: props.row, col: col.name})"
        >
            {{ col.value }}
        </q-td>
    </q-tr>
''')
table.on('cell-click', lambda msg: ui.notify(f'Row "{msg["args"]["row"]["name"]}", column "{msg["args"]["col"]}"'))
```

**Reply by mbolder** [reply]
Brilliant, thank you for the fast reply.

**Reply by j4nSolo** [reply]
I have a problem with this method: 
I use the event callback to modify a previously created ui.dialog, however the javascript behind fails and the whole row disappears.

This is what I can see on the console:

> vue.global.prod.js:1 DOMException: Failed to execute 'insertBefore' on 'Node': The node before which the new node is to be inserted is not a child of this node.
    at insert (http://192.168.0.11:6142/_nicegui/1.4.25/static/vue.global.prod.js:3:34859)
    at k (http://192.168.0.11:6142/_nicegui/1.4.25/static/vue.global.prod.js:3:17228)
    at C (http://192.168.0.11:6142/_nicegui/1.4.25/static/vue.global.prod.js:3:16697)
    at y (http://192.168.0.11:6142/_nicegui/1.4.25/static/vue.global.prod.js:3:16270)
    at ye.fn (http://192.168.0.11:6142/_nicegui/1.4.25/static/vue.global.prod.js:3:22223)
    at ye.run (http://192.168.0.11:6142/_nicegui/1.4.25/static/vue.global.prod.js:1:5983)
    at L.e.update (http://192.168.0.11:6142/_nicegui/1.4.25/static/vue.global.prod.js:3:22468)
    at L (http://192.168.0.11:6142/_nicegui/1.4.25/static/vue.global.prod.js:3:22494)
    at M (http://192.168.0.11:6142/_nicegui/1.4.25/static/vue.global.prod.js:3:20929)
    at F (http://192.168.0.11:6142/_nicegui/1.4.25/static/vue.global.prod.js:3:19195)

Is this somehow messing up with NiceGUI's stack system?

**Reply by falkoschindler** [reply]
@j4nSolo Without seeing your code it's hard to tell what's going on. Vue stacktraces are usually pretty cryptic and don't give much insight about the root cause of the error. Given this thread has been closed a year ago, I suggest to open a new discussion including a minimal executable code example. Thanks!

By the way, this example is pretty old. The last line should be changed to
```py
table.on('cell-click', lambda e: ui.notify(f'Row "{e.args["row"]["name"]}", column "{e.args["col"]}"'))
```

**Reply by j4nSolo** [reply]
Yeah, I know. I adapted it already. Here's the code:

```py
async def cell_clicked(args):
    if args.args['col'] == ATHLETE:
        bib = args.args['row'][BIB]
        athlete_name = args.args['row'][ATHLETE]
        await participant_dialog_content(dialog, race_name_kebab, bib, athlete_name)

datatable.on('cell-click', cell_clicked)

async def participant_dialog_content(dialog, race_name_kebab, bib, athlete_name):
    dialog.clear()
    with dialog:
        with ui.card().classes('gap-0 p-1'):
            with ui.card_section().classes('w-full flex items-end p-2'):
                ui.label(athlete_name)
                ui.space()
                i = ui.icon('close').classes('cursor-pointer').on('click', dialog.close)
            with ui.card_section().classes('w-full p-2'):
                await participant_details(race_name_kebab, bib, False)
                dialog.open()
```

No matter what I do (I tried commenting out portions of the last function), as long as I call the `dialog.clear()` or `ui.card()`, the described behaviour happens.

**Reply by falkoschindler** [reply]
Sorry, given this code snippet, I would need to take it, merge it with other code from this thread, fill the gaps of undefined variables, try to reproduce your problem to finally start debugging. This is pretty inefficient. Therefore I kindly ask you again to provide an executable example. ([Why is it important to provide a minimal reproducible example?](https://github.com/zauberzeug/nicegui/wiki/FAQs#why-is-it-important-to-provide-a-minimal-reproducible-example))
